internal/transactionsubmission/engine/signing/signature_client.go

```diff
@@ -0,0 +1,101 @@
+package signing
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/stellar/go/txnbuild"
+	"golang.org/x/exp/slices"
+
+	"github.com/stellar/stellar-disbursement-platform-backend/db"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/transactionsubmission/engine/preconditions"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/transactionsubmission/utils"
+)
+
+//go:generate mockery --name=SignatureClient --case=underscore --structname=MockSignatureClient
+type SignatureClient interface {
+	NetworkPassphrase() string
+	SignStellarTransaction(ctx context.Context, stellarTx *txnbuild.Transaction, stellarAccounts ...string) (signedStellarTx *txnbuild.Transaction, err error)
+	SignFeeBumpStellarTransaction(ctx context.Context, feeBumpStellarTx *txnbuild.FeeBumpTransaction, stellarAccounts ...string) (signedFeeBumpStellarTx *txnbuild.FeeBumpTransaction, err error)
+	BatchInsert(ctx context.Context, amount int) (publicKeys []string, err error)
+	Delete(ctx context.Context, publicKey string) error
+	Type() string
+}
+
+type SignatureClientType string
+
+const (
+	SignatureClientTypeChannelAccountDB       SignatureClientType = "CHANNEL_ACCOUNT_DB"
+	SignatureClientTypeDistributionAccountEnv SignatureClientType = "DISTRIBUTION_ACCOUNT_ENV"
+	SignatureClientTypeHostAccountEnv         SignatureClientType = "HOST_ACCOUNT_ENV"
+)
+
+func (t SignatureClientType) All() []SignatureClientType {
+	return []SignatureClientType{SignatureClientTypeChannelAccountDB, SignatureClientTypeDistributionAccountEnv, SignatureClientTypeHostAccountEnv}
+}
```

Comment:
My preference for this is not to be a method but a function. To avoid things like `SignatureClientType("").All()`. Since you are not manipulating the type of the instance it makes sense to not be a method. But no need to change it for now since we are widely using this pattern over the codebase.

```go
func SignatureClientTypes() []SignatureClientType {
	return []SignatureClientType{SignatureClientTypeChannelAccountDB, SignatureClientTypeDistributionAccountEnv, SignatureClientTypeHostAccountEnv}
}

// usage 
signatureClientTypes := signing.SignatureClientTypes()
```

Reply:
Done in https://github.com/stellar/stellar-disbursement-platform-backend/pull/174/commits/c1e0cafe9e8216208b774f7b37fedc24d25922f3.

Reply:
I wanted to make it work somewhat like static class methods, but that doesn't work so well with the go syntax.